api/desecapi/tests/test_rrsets.py

```diff
@@ -239,14 +248,155 @@ def test_create_my_rr_sets_empty_payload(self):
         response = self.client.post_rr_set(self.my_empty_domain.name)
         self.assertContains(response, 'No data provided', status_code=status.HTTP_400_BAD_REQUEST)
 
-    def test_create_my_rr_sets_unknown_type(self):
-        for _type in ['AA', 'ASDF']:
-            with self.assertPdnsRequests(
-                    self.request_pdns_zone_update_unknown_type(name=self.my_domain.name, unknown_types=_type)
-            ):
-                response = self.client.post_rr_set(self.my_domain.name, records=['1234'], ttl=3660, type=_type)
+    def test_create_my_rr_sets_canonical_content(self):
+        # TODO fill in more examples
+        datas = [
+            # record type: (non-canonical input, canonical output expectation)
+            ('A', ('127.0.000.1', '127.0.0.1')),
+            ('AAAA', ('0000::0000:0001', '::1')),
+            ('AFSDB', ('02 turquoise.femto.edu.', '2 turquoise.femto.edu.')),
+            ('CAA', ('', '')),
+            ('CDNSKEY', ('', '')),
+            ('CDS', ('', '')),
+            ('CERT', ('', '')),
+            ('CNAME', ('', '')),
+            ('DHCID', ('', '')),
+            ('DLV', ('', '')),
+            ('DS', ('', '')),
+            ('EUI48', ('', '')),
+            ('EUI64', ('', '')),
+            ('HINFO', ('', '')),
+            ('IPSECKEY', ('', '')),
+            ('KEY', ('', '')),
+            ('KX', ('', '')),
+            ('LOC', ('', '')),
+            ('MAILA', ('', '')),
+            ('MAILB', ('', '')),
+            ('MINFO', ('', '')),
+            ('MR', ('', '')),
+            ('MX', ('', '')),
+            ('NAPTR', ('', '')),
+            ('NS', ('', '')),
+            ('NSEC', ('', '')),
+            ('NSEC3', ('', '')),
+            ('OPENPGPKEY', ('', '')),
+            ('PTR', ('', '')),
+            ('RP', ('', '')),
+            ('SPF', ('', '')),
+            ('SRV', ('', '')),
+            ('SSHFP', ('', '')),
+            ('TKEY', ('', '')),
+            ('TLSA', ('3 0001 1 000AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA', '3 1 1 000aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')),
+            ('TSIG', ('', '')),
+            ('TXT', ('"foobar"', 'foobar')),
```

Comment:
I'm pretty sure the canonical presentation format does have quotes

Reply:
The notation in this test is misleading: the [canonical form of the record content](https://tools.ietf.org/html/rfc4034#section-6.2) is in __wire__ format, i.e. no quotes. What is compared here is the presentation format reconstructed by dnspython from canonical wire format. AFAIK, there is no definition of canonical presentation format.

The canonical form of `"foobar"` and `foobar` TXT records is `b'\x06foobar'`. [dnspython uses quotes whenever constructing TXT records from wire data](https://github.com/rthalley/dnspython/blob/5a3df82e3f93c4f480f96f78de8678504282a3e0/dns/rdtypes/txtbase.py#L58), so in above code snipped, there should have been quotes.